sanic/server.py

```diff
@@ -185,16 +188,23 @@ def write_error(self, exception):
             version = self.request.version if self.request else '1.1'
             self.transport.write(response.output(version))
             self.transport.close()
+        except RuntimeError:
```

Comment:
In order to reliably execute `self.transport.close()`,
I think it is better to execute `self.transport.close()` in finally clause.